RFCs/0018-Skip-Default-Extensions.md

```diff
@@ -0,0 +1,42 @@
+# 0018 Skip Default Extensions
+
+## Summary
+This note outlines the proposed changes for adding option for skipping lookup and initialization of default extensions.
+
+## Motivation
+In addition to test adapters provided via TestAdaptersPaths, test platform loads default adapters present in Extensions folder. This folder is present alongside vstest.console.exe. Even for test sources where default extensions can't run/discover tests, we still pass test source assembly to them, this causes both increase adapter lookup, their initialization cost, & adds overhead to execution time.
+
+Example: Default extensions can't run/discover XUnit tests, still testplatform initializes default adapters and pass XUnit test assemblies to them.
+
+Proposed changes are focused on improving performance of test run/discovery by allowing option to skip default extensions.
+
+## Proposed Changes
+1. Design mode clients can skip default extensions by setting TestPlatformOptions.SkipDefaultExtensions as true.
+2. TestRequestManager passes SkipDefaultExtensions flag to TestPlatform via CreateDiscoveryRequest/CreateTestRunRequest.
+3. TestPlatform passes SkipDefaultExtensions flag to ProxyDiscoveryManager/ProxyExecutionManager via initialize method.
+4. ProxyDiscoveryManager/ProxyExecutionManager stores this value and uses it to skip default extensions to be passed to test host whenever test host extensions are initialized.
+
+## API Changes
+1. Adding of field SkipDefaultExtensions in TestPlatformOptions.
+2. TestPlatform's CreateDiscoveryRequest/CreateTestRunRequest will accept TestPlatformOptions as additional argument.
+3. Making TestPlatform internal as it is not meant to be exposed. TestPlatform supports entry point only via command line and translation layer.
+4. IProxyDiscoveryManager and IProxyExecutionManager's Initialize method will accept SkipDefaultExtensions as additional argument.
+
+## Performance Improvement Analysis
```

Comment:
Please provide overall perf gain. The numbers provided currently do not give me the full picture.

Reply:
Provided perf gain numbers. please let know if you need additional details.